### PR TITLE
[DOC] RDoc for ARGF

### PR DESCRIPTION
**Pull request by BurdetteLamar**
I tried this as markdown with the shell passages fenced and marked as `sh` or `bash`, but those passages were not colorized;  should they have been?

### CODE REVIEW

File: io.c [hunk 1]
```diff
@@ -14677,47 +14677,122 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ *     p ARGF.read
+ *
+ * - Command and output:
+ *
+ *     $ echo "Open the pod bay doors, Hal." | ruby t.rb
+ *     []
+ *     "Open the pod bay doors, Hal.\n"
+ *
+ * Each filepath is removed from the array when the corresponding file is accessed;
+ * when all files have been accessed, the array is empty:
```
Comment (peterzhu2118) [inline io.c:14768]: Why is this under the `$stdin Input` section? It seems to be about files and not stdin.

**Comment by nobu** [comment]
I agree that the operations on `ARGV` should be separated from `ARGF`.

File: io.c [hunk 2]
```diff
@@ -14677,47 +14677,122 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ *     ARGV.delete_if {|ele| ele.match(option_regexp) }
+ *     p options
+ *     p ARGV
+ *
+ * - Command and output:
+ *
+ *     $ ruby t.rb --foo bar.txt --baz bat.txt
+ *     ["--foo", "bar.txt", "--baz", "bat.txt"]
+ *     ["--foo", "--baz"]
+ *     ["bar.txt", "bat.txt"]
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply io.c]: Fixed?

File: io.c [hunk 3]
```diff
@@ -14677,47 +14677,122 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ * \Method ARGF#read reads the content of oll the specified files into a single string:
+ *
+ * - \File +t.rb+:
+ *
+ *     option_regexp = /^-/
+ *     ARGV.delete_if {|ele| ele.match(option_regexp) }
+ *     p ARGV
+ *     p ARGF.read
+ *
+ * - \\File +bar.txt+:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply io.c]: Fixed.

File: io.c [hunk 4]
```diff
@@ -14677,47 +14677,122 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ *     p ARGV
+ *
+ * - Command and output:
+ *
+ *     $ ruby t.rb --foo bar.txt --baz bat.txt
+ *     ["--foo", "bar.txt", "--baz", "bat.txt"]
+ *     ["--foo", "--baz"]
+ *     ["bar.txt", "bat.txt"]
+ *
+ * \Method ARGF#read reads the content of oll the specified files into a single string:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply io.c]: Fixed.

File: io.c [hunk 5]
```diff
@@ -14677,47 +14677,122 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ *
+ * - \File +t.rb+:
+ *
+ *     option_regexp = /^-/
+ *     ARGV.delete_if {|ele| ele.match(option_regexp) }
+ *     p ARGV
+ *     p ARGF.read
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply io.c]: Fixed.

File: io.c (hunk 1 above)
Comment (BurdetteLamar) [reply io.c]: Moved.

**Comment by BurdetteLamar** [comment]
> I agree that the operations on `ARGV` should be separated from `ARGF`.

Not sure what is needed for this.  Hope all is well now.

**Comment by nobu** [comment]
I think it is worth to mention that file name `"-"` directs to read from `$stdin`, not `./-` file.

**Comment by BurdetteLamar** [comment]
Converting to draft.  The handling of `'-'` on the command line is completely inadequate, and other things need strengthening as well.

**Comment by BurdetteLamar** [comment]
> I tried this as markdown with the shell passages fenced and marked as `sh` or `bash`, but those passages were not colorized; should they have been?

@peterzhu2118, do you know the answer to this?

**Comment by peterzhu2118** [comment]
This worked for me:

```
 * ```sh
 * echo "hi"
 * ```
```

Screenshot 2024-01-10 at 5 06 32 PM

**Comment by BurdetteLamar** [comment]
Thanks much!
Works for me if unnested and without any prompt (I usually use ' $').
Does not work for me if nested (even without prompt).
Just for the record:  my RDoc version is 6.5.0;  yours?


    On Wednesday, January 10, 2024 at 04:06:51 PM CST, Peter Zhu ***@***.***> wrote:  
 
 


This worked for me:
 * ```sh
 * echo "hi"
 * ```
Screenshot.2024-01-10.at.5.06.32.PM.png (view on web)

**Comment by BurdetteLamar** [comment]
Ook!  Should have said "not in a list" and "in a list," not "unnested" and "nested."

    On Wednesday, January 10, 2024 at 06:47:25 PM CST, Burdette Lamar ***@***.***> wrote:  
 
  Thanks much!
Works for me if unnested and without any prompt (I usually use ' $').
Does not work for me if nested (even without prompt).
Just for the record:  my RDoc version is 6.5.0;  yours?


    On Wednesday, January 10, 2024 at 04:06:51 PM CST, Peter Zhu ***@***.***> wrote:  
 
 


This worked for me:
 * ```sh
 * echo "hi"
 * ```
Screenshot.2024-01-10.at.5.06.32.PM.png (view on web)

**Comment by BurdetteLamar** [comment]
By foolishly and repeatedly replying to this in email (instead of here), I've made a mess of this discussion.  My next push will be very different from what's here now;  @nobu and @peterzhu2118: any objection if I close this and start fresh with a new PR?

**Comment by BurdetteLamar** [comment]
Fenced sh in list still did not work for me.  @peterzhu2118, how about you?

**Comment by BurdetteLamar** [comment]
Again marking this PR as Draft, preliminary to closing it.  (No one has objected to closing.)

**Comment by BurdetteLamar** [comment]
Will open new PR for this.

**Pull request closed without merging**